libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/rest/RestClientTest.java

```diff
@@ -633,6 +639,184 @@ public void testAccessorMethodsPreventResponseStreaming() throws Exception {
         }
     }
 
+    /**
+     * Test for batch request
+     *
+     * Run a batch request that:
+     * - creates an account,
+     * - creates a contact,
+     * - run a query that should return newly created account
+     * - run a query that should return newly created contact
+     *
+     * @throws IOException
+     * @throws JSONException
+     */
+    public void testBatchRequest() throws IOException, JSONException {
+        Map<String, Object> accountFields = new HashMap<String, Object>();
+        String accountName = ENTITY_NAME_PREFIX + System.nanoTime();
+        accountFields.put("Name", accountName);
+        RestRequest firstRequest = RestRequest.getRequestForCreate(TestCredentials.API_VERSION, "account", accountFields);
+
+        Map<String, Object> contactFields = new HashMap<String, Object>();
+        String contactName = ENTITY_NAME_PREFIX + System.nanoTime();
+        contactFields.put("LastName", contactName);
+        RestRequest secondRequest = RestRequest.getRequestForCreate(TestCredentials.API_VERSION, "contact", contactFields);
+
+        RestRequest thirdRequest = RestRequest.getRequestForQuery(TestCredentials.API_VERSION, "select Id from Account where Name = '" + accountName + "'");
+        RestRequest fourthRequest = RestRequest.getRequestForQuery(TestCredentials.API_VERSION, "select Id from Contact where Name = '" + contactName + "'");
+
+
+        // Build batch request
+        RestRequest batchRequest = RestRequest.getBatchRequest(TestCredentials.API_VERSION, false, new RestRequest[]{firstRequest, secondRequest, thirdRequest, fourthRequest});
+
+        // Send batch request
+        RestResponse response = restClient.sendSync(batchRequest);
+
+        // Checking response
+        JSONObject jsonResponse = response.asJSONObject();
+
+        checkKeys(jsonResponse, "hasErrors", "results");
+        assertFalse("Batch had errors", jsonResponse.getBoolean("hasErrors"));
+        JSONArray jsonResults = jsonResponse.getJSONArray("results");
+        assertEquals("Wrong number of results", 4, jsonResults.length());
+        assertEquals("Wrong status for first request", 201, jsonResults.getJSONObject(0).getInt("statusCode"));
```

Comment:
Not a big deal, but maybe use `HttpURLConnection.HTTP_CREATED` instead of `201` and `HttpURLConnection.HTTP_OK` instead of `200`, so it's easier to read?